jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -504,25 +505,40 @@ private void complete(Request baseRequest) {
       env.waitForAllApiCallsToComplete(VmRuntimeUtils.MAX_REQUEST_THREAD_API_CALL_WAIT_MS);
     }
 
-    void handleAbandonedTxns(Collection<Transaction> txns) {
-      // TODO(user): In the dev appserver, capture a stack trace whenever a
-      // transaction is started so we can print it here.
-      for (Transaction txn : txns) {
+    void handleAbandonedTxns() {
+      if (getActiveTransactions != null) {
         try {
-          logger.warning(
-              "Request completed without committing or rolling back transaction with id "
-                  + txn.getId()
-                  + ".  Transaction will be rolled back.");
-          txn.rollback();
-        } catch (Exception e) {
-          // We swallow exceptions so that there is no risk of our cleanup
-          // impacting the actual result of the request.
+          Object txns = getActiveTransactions.invoke(contextDatastoreService);
+
+          if (txns instanceof Collection) {
```

Comment:
I prefer having at least a null check as otherwise we are assuming details of getActiveTransactions.   I prefer a instanceof check as then it gives me both a null check and an assert of the returned type.